filebeat/docs/how-does-filebeat-work.asciidoc

```diff
@@ -0,0 +1,76 @@
+== How does Filebeat work?
+
+This page intends to explain what the key building blocks of Filebeat are and how they work together. The goal is that the available configuration options can be better understood on what they do.
+
+=== What is an prospector?
+
+Filebeat currently supports two `prospector` types: `log` and `stdin`. Each prospector type can be defined multiple times. The responsibility of the prospector is to find all sources it should read from. In case of the log input type it is to find all files on the drive based on the defined paths and start a harvester for each file that was found. Each prospector looks as following:
+
+```
+filebeat.prospectors:
+- input_type: log
+  paths:
+    - /var/log/*.log
+    - /var/path2/*.log
+```
+
+The log prospector is responsible to find all files which were defined under the paths above and check, if for each file a harvester has to be started, if one is already running or the file can be ignored because of some configuration options. Each prospector runs in its own go routine and is responsible to manage the harvesters. New files are only picked up, if the offset / size of the file changed since the harvester was closed.
```

Comment:
Would probably be better to avoid "harvester" in this paragraph, since it's not yet introduced.